utilities/test_suite/HIP/Tensor_hip_audio.cpp

```diff
@@ -0,0 +1,258 @@
+    if(testCase == 3)
+        maxDstChannels = 1;
+    set_audio_descriptor_dims_and_strides(dstDescPtr, batchSize, maxDstHeight, maxDstWidth, maxDstChannels, offsetInBytes);
+
+    // set buffer sizes for src/dst
+    iBufferSize = (Rpp64u)srcDescPtr->h * (Rpp64u)srcDescPtr->w * (Rpp64u)srcDescPtr->c * (Rpp64u)srcDescPtr->n;
+    oBufferSize = (Rpp64u)dstDescPtr->h * (Rpp64u)dstDescPtr->w * (Rpp64u)dstDescPtr->c * (Rpp64u)dstDescPtr->n;
+
+    // allocate hip buffers for input & output
+    Rpp32f *inputf32 = (Rpp32f *)calloc(iBufferSize, sizeof(Rpp32f));
+    Rpp32f *outputf32 = (Rpp32f *)calloc(oBufferSize, sizeof(Rpp32f));
+
+    void *d_inputf32, *d_outputf32;
+    CHECK(hipMalloc(&d_inputf32, iBufferSize * sizeof(Rpp32f)));
+    CHECK(hipMalloc(&d_outputf32, oBufferSize * sizeof(Rpp32f)));
+
+    // allocate the buffers for audio length and channels
+    Rpp32s *srcLengthTensor, *channelsTensor;
+    CHECK(hipHostMalloc(&srcLengthTensor, batchSize * sizeof(Rpp32s)));
+    CHECK(hipHostMalloc(&channelsTensor, batchSize * sizeof(Rpp32s)));
+
+    // allocate the buffers for src/dst dimensions for each element in batch
+    RpptImagePatch *srcDims = (RpptImagePatch *) calloc(batchSize, sizeof(RpptImagePatch));
+    RpptImagePatch *dstDims = (RpptImagePatch *) calloc(batchSize, sizeof(RpptImagePatch));
+
+    // allocate the buffer for srcDimsTensor
+    Rpp32s *srcDimsTensor;
+    CHECK(hipHostMalloc(&srcDimsTensor, batchSize * 2 * sizeof(Rpp32s)));
+
+    Rpp32f *detectedIndex = nullptr, *detectionLength = nullptr;
+    if(testCase == 0)
+    {
+        CHECK(hipHostMalloc(&detectedIndex, batchSize * sizeof(Rpp32f)));
+        CHECK(hipHostMalloc(&detectionLength, batchSize * sizeof(Rpp32f)));
+    }
+
+    // run case-wise RPP API and measure time
+    rppHandle_t handle;
+    hipStream_t stream;
+    CHECK(hipStreamCreate(&stream));
+    rppCreateWithStreamAndBatchSize(&handle, stream, batchSize);
+
+    int noOfIterations = (int)audioNames.size() / batchSize;
+    double maxWallTime = 0, minWallTime = 500, avgWallTime = 0;
+    string testCaseName;
+    printf("\nRunning %s %d times (each time with a batch size of %d images) and computing mean statistics...", func.c_str(), numRuns, batchSize);
+    for (int iterCount = 0; iterCount < noOfIterations; iterCount++)
+    {
+        // read and decode audio and fill the audio dim values
+        read_audio_batch_and_fill_dims(srcDescPtr, inputf32, audioFilesPath, iterCount, srcLengthTensor, channelsTensor);
+        CHECK(hipMemcpy(d_inputf32, inputf32, iBufferSize * sizeof(Rpp32f), hipMemcpyHostToDevice));
+        for (int perfRunCount = 0; perfRunCount < numRuns; perfRunCount++)
+        {
+            double startWallTime, endWallTime;
+            double wallTime;
+            switch (testCase)
+            {
+                case 2:
+                {
+                    testCaseName = "pre_emphasis_filter";
+                    float coeff[batchSize];
```

Comment:
Create a pinned memory for coeff outside this loop
and free the memory at the end of this function

Please refer to previous comment i added in glitch/pixelate PR regarding how to do this

Reply:
Done